tests/integrationtest/r/ddl/default_as_expression.result

```diff
@@ -955,3 +955,62 @@ child	CREATE TABLE `child` (
   CONSTRAINT `fk_1` FOREIGN KEY (`cc`) REFERENCES `parent` (`c`) ON DELETE CASCADE
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
 drop table parent, child;
+CREATE TABLE jd1 (id SERIAL, j JSON DEFAULT (JSON_OBJECT("foo", "bar")));
+SHOW CREATE TABLE jd1;
+Table	Create Table
+jd1	CREATE TABLE `jd1` (
+  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+  `j` json DEFAULT (json_object(_utf8mb4'foo', _utf8mb4'bar')),
+  UNIQUE KEY `id` (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+INSERT INTO jd1(id) VALUES(1);
+TABLE jd1;
+id	j
+1	{"foo": "bar"}
+CREATE TABLE jd2 (id SERIAL, j JSON DEFAULT (JSON_ARRAY("foo", "bar")));
+SHOW CREATE TABLE jd2;
+Table	Create Table
+jd2	CREATE TABLE `jd2` (
+  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+  `j` json DEFAULT (json_array(_utf8mb4'foo', _utf8mb4'bar')),
+  UNIQUE KEY `id` (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+INSERT INTO jd2(id) VALUES(1);
+TABLE jd2;
+id	j
+1	["foo", "bar"]
+CREATE TABLE jd3 (id SERIAL, j JSON DEFAULT (JSON_QUOTE("foobar")));
+SHOW CREATE TABLE jd3;
+Table	Create Table
+jd3	CREATE TABLE `jd3` (
+  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+  `j` json DEFAULT (json_quote(_utf8mb4'foobar')),
+  UNIQUE KEY `id` (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+INSERT INTO jd3(id) VALUES(1);
+TABLE jd3;
+id	j
+1	"foobar"
+CREATE TABLE jd4 (id SERIAL);
+SHOW CREATE TABLE jd4;
+Table	Create Table
+jd4	CREATE TABLE `jd4` (
+  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+  UNIQUE KEY `id` (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+ALTER TABLE jd4 ADD COLUMN j JSON;
+SHOW CREATE TABLE jd4;
+Table	Create Table
+jd4	CREATE TABLE `jd4` (
+  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+  `j` json DEFAULT NULL,
+  UNIQUE KEY `id` (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+ALTER TABLE jd4 MODIFY COLUMN j JSON DEFAULT (JSON_QUOTE("foobar"));
+SHOW CREATE TABLE jd4;
+Table	Create Table
+jd4	CREATE TABLE `jd4` (
+  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+  `j` json DEFAULT (json_quote(_utf8mb4'foobar')),
+  UNIQUE KEY `id` (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
```

Comment:
Maybe we need to add more tests like:
* [modify column, set default value, add index, add column](https://github.com/pingcap/tidb/blob/a44d4090645d32d3b6b68b866b828b55889321b4/tests/integrationtest/t/ddl/default_as_expression.test#L42)
* [test generated column and expression ](indexhttps://github.com/pingcap/tidb/blob/a44d4090645d32d3b6b68b866b828b55889321b4/tests/integrationtest/t/ddl/default_as_expression.test#L409)
* [partition table and global ](indexhttps://github.com/pingcap/tidb/blob/a44d4090645d32d3b6b68b866b828b55889321b4/tests/integrationtest/t/ddl/default_as_expression.test#L439)